src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -165,8 +134,11 @@ public IReadOnlyList<SourceRepository> GlobalPackageFolderRepositories
 
                     // Read package folders from settings
                     var pathContext = NuGetPathContext.Create(Settings);
-                    var folders = new List<string>();
-                    folders.Add(pathContext.UserPackageFolder);
+                    var folders = new List<string>
```

Comment:
you can get slightly better perf by giving this list an initial capacity of the size of pathContext.UserPackageFolder + pathContext.FallbackPackageFolder so it doesn't have do a copy of the whole list when you do AddRange later.

you can also initialize the list by adding pathContext.FallbackPackageFolders first, and then add pathContext.UserPackageFolder.